core/src/main/kotlin/plugability/DokkaPlugin.kt

```diff
@@ -60,7 +60,10 @@ inline fun <reified P : DokkaPlugin, reified E : Any> P.query(extension: P.() ->
     context?.let { it[extension()] } ?: throwIllegalQuery()
 
 inline fun <reified P : DokkaPlugin, reified E : Any> P.querySingle(extension: P.() -> ExtensionPoint<E>): E =
-    context?.single(extension()) ?: throwIllegalQuery()
+    querySingleOrNull(extension) ?: throwIllegalQuery()
+
+inline fun <reified P : DokkaPlugin, reified E : Any> P.querySingleOrNull(extension: P.() -> ExtensionPoint<E>): E? =
+    context?.single(extension())
```

Comment:
This is not working as intended. `context.single` should never return null. What can be null here is the context itself.

Reply:
I am not a fan of this, since it gives you a false hope that whenever you want to resolve something that isn't there null will be returned. Instead we will throw an exception. 

A possible solution would be to add a `singleOrNull` to `DokkaContext`, but i am not entirely sure it is a good idea because of stability reasons. Also a single nullable value doesn't reflect 3 states that could occur: no element, 1 element, multiple elements

Reply:
Rewerted.